### ISSUE TITLE
ion-select component

### ISSUE THREAD

**Issue by adamdbradley** [driftyco/ionic2#890]
Historically the `<select>` element has been nothing but trouble for Ionic. The `select` and `option` elements both have different behaviors on different browsers, browser versions and operating systems. Trying to bring focus/blur to the select element, having the webapp scroll the element into view, preventing default, etc. are all challenges to do cross platform.

And the strangest part is, the `select` element concept really isn't "used" within native apps. Forget the fact that ionic runs on the web, native apps themselves do not have this same concept of something like this:
![image](https://cloud.githubusercontent.com/assets/452425/12182426/cf719526-b54d-11e5-8130-649b2080b326.png)

Native iOS apps would rather you do something like an actionsheet or alert, and Android does a good job of bringing up dialogs with options:
![image](https://cloud.githubusercontent.com/assets/452425/12182534/5e59ce48-b54e-11e5-8907-fbb595218791.png)

Next, it's well known HTML inputs are just taken as they are, and they are very difficult to style without ugly ugly hacks (and hacks that pretty much don't work on firefox anyways). This is where the web-component model kicks in, allowing devs to build their own inputs, and what I think we should do too.

So basically discourage the use of the `<select>` element all together, and instead build a `<ion-select>` which is the same concept as `<ion-checkbox>`, `<ion-radio>`, `<ion-toggle>`. Like those components we can avoid all the pitfalls of HTML inputs. It could be setup the same way a normal select would, with child options, but when tapping the ion-select it would bring up an alert with a scrollable list of the options.

Thoughts? @mlynch @brandyscarney @mhartington @drewrygh @tlancina 

Related: #886

**Comment by mhartington** [comment]
All for it. I know there are some example

http://codepen.io/bianchimro/pen/epYYQO?editors=101

Where dev's have hacked the modal used that instead. 
This could be something we look into where we could use the modal component or action sheet for the styling and animation. 

something like 

```html
<ion-list>
  <ion-select>
    <ion-option *ngFor="option of options">{{option.name}}</ion-option>
  </ion-select>
</ion-list>
```

And this would just animate up like a native select element.

**Comment by adamdbradley** [comment]
Love it. Really most of the select magic can be added to `Alert`, by allowing an `options` array to be passed to it, just like how alerts can have an array of `buttons`. So this would add some power to Alerts, and the `ion-select` component would really just be a thin layer around firing up an alert with the given ion-select/option markup, and allowing it to style nicely within a list/card.

**Comment by Luchillo** [comment]
Cool, selects doesn't fit very well in mobile, and radio buttons in a popup feels more native, and also are easier to style, like this idea, is the `ion-select` already available in alpha.48?

**Comment by mhartington** [comment]
This is something we're still discussing so it's not available yet. Though any thoughts and comments are welcomed.

**Comment by Luchillo** [comment]
Totally like the idea, never tough about replacing that ugly select with radio buttons in a popup, it actually looks cooler.

**Comment by dmackerman** [comment]
Love the idea of turning "select" into a modal -> select options from a list. That's really the most common pattern. On iOS, a lot of "select" options are actually nested in separate view, which is just a list view of more options. Tap to select, transitions back to the previous view.

**Comment by richardshergold** [comment]
I love this discussion.  I'm using select in my two Ionic 2 apps and having to hack them a little bit and use trial and error to get a reasonable result iOS and Android.  It seems to be very hit and miss how they will behave depending on the platform and parent container i.e a list or a card or whatever.  So anything to make that better would be very welcome.

**Comment by astec** [comment]
@adamdbradley, I suggest you consider support for multi-select from beginning - it's such a common need ([*an HTML standard*](http://www.w3schools.com/tags/att_select_multiple.asp)) and always a pain to implement. Would be cool if it works in Ionic out of the box. For me personally it would be one of a core features.

Maybe in basic version just comma separated list of values + number of not fitted if needed.

Can/should be a separate `ion-multi-select` control.

===================
**Makes**
[ Audi, Volvo + 3 more... ▼ ]

**Comment by mhartington** [comment]
@astec or multiple selections could be a property. 

ion-select[multiple]

**Comment by Luchillo** [comment]
Also the multi select could be like the modal with radios talked above but instead of radios, use switches to control which ones would be selected.

**Comment by astec** [comment]
@mhartington, the *multiple* can be a property as in HTML but I myself would make it separate controls so there is a clear strongly typed difference in value(s).

```
class Select:
  value: string
class MultiSelect:
  values: string[]
```
Not sure if it should be `string` or `any` though. They can be as something like:
```
class AbstractSelect:
  value: string // Created from either selectedValue or selectedValues
class SingleSelect(AbstractSelect):
  selectedValue: any
class MultiSelect(AbstractSelect):
  selectedValues: any[]
```

**Comment by adamdbradley** [comment]
The multi-select feature is a good idea and actually won't be difficult to implement. The select will use an alert with radio buttons, and the multi-select will use an alert with checkboxes. I'm torn if it should be a new component or not, or reuse the same component and have an attribute. We'll hash it over more, thanks.

**Comment by mhartington** [comment]
I think what ever one way we go, we should make sure that we can work with all of ng2's forms directives

**Comment by adamdbradley** [comment]
Thinking more on it I think everyone has a great point about having two different components. You wouldn't want a component to sometimes give a single value, and sometimes give multiple values. Consider it done `<ion-multi-select>`.

**Comment by astec** [comment]
@adamdbradley :thumbsup:

**Comment by richardshergold** [comment]
@adamdbradley excellent - I look forward to seeing this in alpha49 !

**Comment by Luchillo** [comment]
I still think it will look cooler if the multi select make use of switches instead of checkboxes, maybe put an option for it?

**Comment by adamdbradley** [comment]
Just as an update, @mlynch convinced me of `<ion-select multiple>` for multiple selects. Additionally using the same component will reduce the amount of code and overhead.

**Comment by adamdbradley** [comment]
The functionality will be in the next release. Also note, we still need to touch up the CSS to make sure it "fits" within iOS and Material Design. cc @bensperry

**Comment by bensperry** [comment]
What can I do to help, Sir Bradley?

**Comment by richardshergold** [comment]
@adambradley that would be fantastic !  I was only joking about it being in Alpha49 :-)
